base/range.jl

```diff
@@ -140,7 +140,20 @@ abstract type AbstractRange{T} <: AbstractArray{T,1} end
 RangeStepStyle(::Type{<:AbstractRange}) = RangeStepIrregular()
 RangeStepStyle(::Type{<:AbstractRange{<:Integer}}) = RangeStepRegular()
 
-convert(::Type{T}, r::AbstractRange) where {T<:AbstractRange} = r isa T ? r : T(r)
+AxesStartStyle(::Type{<:AbstractRange}) = AxesStart1()  # opt-out of AxesStart1 for "weird" range types
```

Comment:
In some ways I'd prefer "opt in" on this trait, but I think it's too breaking. So I think it's better to mark `AbstractRange` types that violate the expectation of having unit indexing.